src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -707,6 +708,25 @@ private void ParseHelper(string[] args)
                         break;
                     }
                 }
+
+                else if (MatchSwitch(switchKey, "settingsfile", "settings") )
+                {
+                    ++i;
+                    if (i >= args.Length)
+                    {
+                        WriteCommandLineError(
+                            CommandLineParameterParserStrings.MissingSettingsFileArgument);
+                        break;
+                    }
+                    string configFile = args[i];
+                    if (!System.IO.File.Exists(configFile))
```

Comment:
The validation of the file path should be similar to how `-File` does: normalize path separator and then get full path
See: https://github.com/dantraMSFT/PowerShell/blob/f587fd06d6046859ca1a9fe3470d5f9615dc5e01/src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs#L928

Reply:
And, shall we check the extension of the file path to be `json`, like `-File` does?
```
PS:1> powershell -File F:\tmp\decode.json
Processing -File 'F:\tmp\decode.json' failed because the file does not have a '.ps1' extension. Specify a valid Windows PowerShell script file name, and then try again.
```

Reply:
I'm not aware of any requirement for json files to have the extension .json.  Is this really something we want to enforce?

Reply:
Validation fixed